Source/Orts.Simulation/MultiPlayer/Message.cs

```diff
@@ -3018,7 +3019,11 @@ public class MSGSignalStatus : Message
         static byte[] preState;
         static SortedList<long, SignalHead> signals;
         public bool OKtoSend = false;
+        bool SendEverything;
         static byte[] signalsStates;
+        int changedTextStates = 0;
+        static string[] signalTextStates;
+        static string[] preTextState;
```

Comment:
So unfortunately, we have to maintain backwards compatibility in the multiplayer protocol. (Yes, this sucks.) The reason is because there is a [public relay server](http://www.tsimserver.com/ORFiles031205/ServerInfo.html), whose source code has not been made public and whose protocol is fixed, that is used by many Open Rails players. We will get complaints if Open Rails can no longer connect to this server.

I am not familiar with this section of the multiplayer code, so maybe there's no issue, but it's something to be aware of FYI.

Reply:
Today I participated in a MP session using the public server and there were not more crashes than usual. We have to test it again, with more signals using text aspects, to see if it works well. I think the server just broadcasts the message, as it is sent as a base64 compressed string, and there is no point in decoding it there.

Reply:
I take it all went well.